lib/services/sorting/post_sorting_service.dart

```diff
@@ -0,0 +1,59 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:collection/collection.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/services/sorting/post_sort_option.dart";
+
+class PostSortingService {
+  /// This method sorts the [posts] according to the [option]
+  /// and the [position] of the user. It also allows to put some
+  /// posts on top with the [putOnTop] set.
+  /// For instance, let's say that we have a list [A, b, c, D, e]
+  /// randomly shuffled that we sort by alphabetical order. If
+  /// putOnTop contains [A, D, K], then the reuslt will be
+  /// [A, D, b, c, e].
+  List<PostFirestore> sort(
+    List<PostFirestore> posts,
+    PostSortOption option,
+    GeoPoint position, {
+    Set<PostFirestore> putOnTop = const {},
+  }) {
+    int defaultComparator(a, b) {
```

Comment:
I just believe we can specify the parameter types in the function here and maybe rename them.
```suggestion
    int defaultComparator(PostFirestore postA, PostFirestore postB) {
```

Reply:
Fixed in d36c080e5f63c82ff79a2853138390e7856148d8.